make sure domain strings are only sent once

derekwbrown opened the pull request:
### What does this PR do?

consolidates all domain strings into a single list, so that duplicates are not sent

### Motivation

more efficient packaging of submission data


### Checklist
<!-- Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields. -->

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.

derekwbrown committed:
make sure domain strings are only sent once

derekwbrown committed:
make sure domain strings are only sent once

derekwbrown committed:
gofmt

derekwbrown committed:
gofmt

derekwbrown committed:
update to latest agent-payload

derekwbrown committed:
update to latest agent-payload

derekwbrown committed:
Merge branch 'main' into db/dedup-domains

ahmed-mez committed:
[KSMCore] Fix label joins (#8508)

L3n41c committed:
Add a single entrypoint for the agent docker image (#8490)

ahmed-mez committed:
[SB] Add readsecret.sh to core Agent (#8564)

vboulineau committed:
Fix CLC detection in case clusterchecks provider is set multiple times (#8362)

remeh committed:
[aggregator/ckey] implement a faster hash to reduce CPU and RAM usage (#8529)

* [aggregator/ckey] implement a hash which doesn't need sorted tags

The highest CPU usage in the CPU profiles of highly loaded DogStatsD server was
the sorting of the tags to generate reproducible context keys.

This commit implements a hash which does not need to sort the tags to generate
reproducible context keys: it relies on XORing parts of the hash together instead
of concatening bytes before hashing the whole bytes buffer. In order for this to
work, a fast deduplication algorithm had to be written since duplicated tags would
make the hash computation wrong because of the XOR usage.

Using this implementation, the DogStatsD server will use less CPU to process more
metrics, and thus, it will use less RAM since it won't have to buffer as much as
before.

* Apply suggestions from code review

Co-authored-by: Vickenty Fesunov <vickenty@users.noreply.github.com>

* [aggregator/ckey] address feedback.

* [aggregator/ckey] add a note on a possible enhancement.

Co-authored-by: Vickenty Fesunov <vickenty@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/itchyny/gojq from 0.12.3 to 0.12.4 (#8465)

Bumps [github.com/itchyny/gojq](https://github.com/itchyny/gojq) from 0.12.3 to 0.12.4.
- [Release notes](https://github.com/itchyny/gojq/releases)
- [Changelog](https://github.com/itchyny/gojq/blob/main/CHANGELOG.md)
- [Commits](https://github.com/itchyny/gojq/compare/v0.12.3...v0.12.4)

---
updated-dependencies:
- dependency-name: github.com/itchyny/gojq
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

olivielpeau committed:
[collector] Use error-level log for check instance load error (#8294)

Follow up to 12b45ae8218f878c04c000af02dac415ba03744d

albertvaka committed:
Move Linux kitchen tests from US North to other regions (#8567)

This is a workaround. In US North Azure VMs, sudo complains with:

sudo: unable to resolve host <hostname>: Name or service not known

Which breaks our parsing of the agent status because we call `sudo datadog-agent status -j 2>&1`

The better fix would be to not redirect stderr to stdout.

vboulineau committed:
Move E2E tests to kind (#8545)

truthbk committed:
[circleci] address python linting issue adding env vars (#8572)

khewonc committed:
Report container inclusion/exclusion errors in agent status (#8481)

* Display container filtering errors in agent status

* Add release note

* Fix missing log filter exclusions, remove separate FilterErrors struct, github comments

* Fix formatting

L3n41c committed:
Migrate docker publication GitLab jobs to public-images (#8134)

Co-authored-by: Kylian Serrania <kylian.serrania@datadoghq.com>
Co-authored-by: Albert Vaca Cintora <albert.vaca@datadoghq.com>

gh123man committed:
Log bytes and lines processed by file tailer when the tailer stops (#8374)

Add the # of bytes and # of raw lines processed by a tailer when it stops.

p-lambert committed:
[networks] Add support for sendfile(2) traffic (#8524)

ogaca-dd committed:
Fix win service shutdown (#8436)

* Fix the error "The DataDog Agent service terminated unexpectedly." when restarting a Windows host.

gbbr committed:
pkg/trace/sampler: add an LRU for the service catalog (#8559)

This change makes the service catalog an LRU cache with a maximum
entries of 5000.

vickenty committed:
[profiling] expose more profiling options, take 2 (#8558)

* [profiling] remove unused apiKey parameter

dd-trace-go/profiler does not require an api key, because it always
sends to the trace agent (unless WithAgentlessUpload option is used,
which our code does not).

* [profiling] expose mutex and block options

Disable mutex profiling by default, and make it configurable instead.

Add configuration options to process and system-probe agents to
configure mutex and block profiling.

* [trace] update call to profiling.Start

profiling.Start api has changed in the last commit, update this new call
as well.

ahmed-mez committed:
[KSMCore] kube_cronjob tag on job metrics (#8576)

vickenty committed:
[logs] Add support for new intake api (#8555)

* [logs] Add support for new intake api

Add support for new version 2 intake API used for logs and events.
New API supports additional parameters to differentiate requests: a
track type, which is part of a url, and an optional `dd-protocol`
header.

Each component that interacts that interacts with the intake, was
assigned an individual type; this is passed via various routes into the
endpoint construction.

Whether new API is used or not is controlled by a new hidden
`use_v2_api` configuratio option (available, like other options, for
each agent component separately).

Additional endpoints inherit the api version setting from the parent
endpoint, but it can be overridden if necessary.

* [logs] use correct protocol for serverless agent

* [security-agent] use correct track for intake

Security agent interacts with two different intake endpoints, and needs
to use a specific track type with each.

* [epforwarder] update track type for dbm metrics

* [logs] use net/url to build the destination url

lebauce committed:
Build syscall tester statically (#8557)

brycekahle committed:
Update to go 1.15.13 (#8540)

Co-authored-by: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

juliogreff committed:
[DCA] Introduce use_component_status for kube_apiserver check (#8577)

Previously, the kube_apiserver_controlplane used ComponentStatus to
report control plane components' liveness. This has been deprecated in
[Kubernetes 1.19](https://github.com/kubernetes/kubernetes/pull/93570)
and will be removed at some point in the future.

To remediate that, we're following the recommendation in the deprecation
notice to use the API Server's health endpoint instead. This change also
removes the `component` tag in this service check, as it no longer
reports separate components, and just the API server itself.
Per-component service checks will eventually be available through the
kube_controller_manager and kube_scheduler checks themselves.

safchain committed:
Add agent version to event log (#8568)

julien-lebot committed:
Use custom Python 2 interpreter for Windows (#8566)

khewonc committed:
Apply `max_returned_metrics` from prometheus annotations to openmetrics check (#8580)

* Set max_returned_metrics on openmetrics check if configured on prometheus annotations

brycekahle committed:
Store HTTP latencies as nanoseconds represented by float64 (#8562)

paulcacheux committed:
[runtime-security] add selinux event probes (#8370)

* [runtime-security] add basic kprobe on vfs_write

* [runtime-security] improve kprobe and enable loading

* [runtime-security] hook directly selinux functions

* [runtime-security] define selinux event in userspace code

* [runtime-security] enable selinux event tracing

* [runtime-security] generate fields for selinux event struct

* [runtime-security] add really basic test for selinux events

* [runtime-security] add file information to selinux events

* [runtime-security] add file serializer for selinux event

* [runtime-security] improve probe declaration and add more selinux probes

* [runtime-security] un-force always loading of selinux probes

* [runtime-security] fix dentry resolving issues in selinux event probes

* [runtime-security] improve tests for selinux events

* [runtime-security] export write buffer in selinux events

* [runtime-security] improve selinux events tests with bool values

* [runtime-security] improve selinux events tests

* [runtime-security] add missing comment on SELinuxRawBufCapacity

* [runtime-security] add more selinux probes

* [runtime-security] fix timeout error when probing an unexisting function

* [runtime-security] change SELinuxEvent buf size, to fill padding

* [runtime-security] simplify SELinux probes

* [runtime-security] add event kind for SELinux event, in order to differentiate them

* [runtime-security] rename function

* [runtime-security] edit SELinux fields to match discussed taxonomy

* [runtime-security] fix selinux tests

* [runtime-security] add selinux event serializers

* [runtime-security] add information about changes in SELinux events

* [runtime-security] validate selinux event json schemas

* [runtime-security] simplify selinux event kind enum

* [runtime-security] fix go vet errors

* [runtime-security] fix comment misspelling

* [runtime-security] add change field to enforce status events

* [runtime-security] fix SELinux test flakiness and state restoration

* [runtime-security] update runtime security hash

* [runtime-security] add reno release note

* [runtime-security] fix selinux tests to work with new race detector

* [runtime-security] remove "changed" for bool selinux events

* [runtime-security] remove "changed" for enforce status selinux event

* [runtime-security] improve selinux write value collection

* [runtime-security] simplify bool values resolving

* [runtime-security] keep track of enforce status in a bpf map

* [runtime-security] snapshot the starting state of SELinux status at start

* [runtime-security] fix go vet errors

* [runtime-security] fix review points + run clang-format on `selinux.h`

julien-lebot committed:
Build dockerized Windows Agent 6 for testing Python 2 (#8488)

safchain committed:
Make sure to not use external deps in model (#8582)

safchain committed:
Report policies versions as part of the running metric (#8569)

AlexandreYang committed:
[corechecks/snmp] Set default oid_batch_size to 5 (#8575)

vboulineau committed:
Use Env AD to choose which providers to trigger in different components (tagger, hostname, host tags...) (#8521)

* Use Env AD to choose which providers to trigger in different components (tagger, hostname, host tags...)

* Add regexp support to exclude features

julien-lebot committed:
Add libdatadog-agent-two to zip (#8589)

vboulineau committed:
Fix missing .node.count metric in KSM Core (#8581)

coignetp committed:
Update package tested version (#8588)

Commit:
[orchestrator] update follower case (#8579)

This PR updated the status output for the DCA follower for the orchestrator explorer (live containers) output. If the status command is called from the follower a new message shows that the collection only happens on the leader.

julien-lebot committed:
Revert "Use custom Python 2 interpreter for Windows (#8566)" (#8593)

This reverts commit 13c19c9a25f9f19e37c43eeb8216f81c06a0d61a.

ahmed-mez committed:
[KSMCore] Add DD_TAGS (#8587)

vickenty committed:
[jmxfetch] Upgrade to 0.44.2 (#8592)

peterg17 committed:
pkg/trace/api: add orchestrator tags to profiles in Fargate environments (#8523)

* Add tags to help identify profiles from Fargate.
* Add fargate test for profiles.go.

ISauve committed:
Update CODEOWNERS (#8594)

Makes agent-network the owners of pkg/config/system_probe.go

p-lambert committed:
[system-probe] Terminate gracefully when system-probe is not enabled (#8578)

* [system-probe] Terminate gracefully when system-probe is not enabled

* [networks] Add back-off for Linux

* [networks] Change status to stopped before exiting

alexbarksdale committed:
Add options support for the SQL obfuscator and update CGo bindings (#8556)

* [Fix] Invalid memory address or nil pointer test panic

* [Add] SQL obfuscation config + easyjson

* [Add] Options argument for C bindings + signature updates

* [Add] CGo bindings: consume obfuscator options

* [Add] Obfuscation options arg

* [Add] SQL obfuscation config deserialization method test + benchmarks

* [Add] Changelog

* [Update] Pass options directly rather than using receiver struct

* [Update] Import order + small nit fix

* [Fix] Attempt to fix failing config test cases

* [Update] Provide obfuscation methods with & without options

* [Revert] Test case back to original

* [Update] Var name cleanup

* [Update] Test and variable cleanup + prefix release note

* [Update] Remove WithOptions methods and update existing signatures

* [Add] Provide feature flag value as option

* [Update] Move options from config to obfuscator

* [Add] Provide feature flag value

* [Add] WithOption methods

* [Fix] Incorrect method use

* [Update] Inline default config

* [Update] Release note

* [Update] Clarity

brycekahle committed:
Bugfixes to windows npm (#8595)

Found during dev of https://github.com/DataDog/datadog-agent/pull/8585

ISauve committed:
[system-probe] Automatically download kernel headers (1) (#7871)

* [system-probe] Automatically download kernel headers

* Address PR comments

* Refactor header search/download logic

* Add kernel header fetching telemetry

* Fix TestGetKernelHeaders & gofmt issues

* Update agent-payload dependency + re-generate licenses

* Combine getDefaultHeaderDirs + getDownloadedHeaderDirs

* Only clean default header download directory

* Remove delete header download directory logic

* Remove double quotes from license file

* Re-generate license file

* Remove duplicate code + fix truncated comment

* Validate downloaded headers

celenechang committed:
[containers] improve handling of CPU metrics when missing cgroup files (#8398)

* [containers] submit -1 for cpu metrics when nil

truthbk committed:
[changelog] adding 7.29.1 details

vickenty committed:
Adding 7.30.0-rc.1 to the release.json (#8600)

* [release] cleanup release.json after 7.29.0

Remove old RC entries and add the final release

* [release] add release entry for 7.30.0-rc.1

* [release] add release note for omnibus repos

celenechang committed:
Revert "[containers] improve handling of CPU metrics when missing cgroup files (#8398)" (#8601)

This reverts commit adbca5f33d40e0afe72a10d5673de06cfd1347eb.

L3n41c committed:
Fix docker image deployment (#8604)

vickenty committed:
[release] add release entry for 7.30.0-rc.2 (#8605)

derekwbrown committed:
make dns by querytype encoding changes configurable (#8586)

* make dns by querytype encoding changes configurable

* review feedback

* change config to more readable map dereference

* Fixes based on reviews
- set bydomain/byquerytype flag in system probe config (not main config)
- when multiple query types are enabled, combine them when we're not
  returning by domain/by query type
- make sure to build the domain list independent of how dns stats are
  returned

* remove missed <break> from pvs. fix

* last review feedback

Co-authored-by: Olivier Vielpeau <olivier.vielpeau@datadoghq.com>

vboulineau committed:
Do not retry tagger collector when feature is disabled (#8610)

vickenty committed:
[release] add release entry for 7.30.0-rc.3 (#8612)

vboulineau committed:
Fix condition to use Kubelet hostname provider (#8616)

truthbk committed:
[changelog] update with corrections for 7.29.1

maxday committed:
[Serverless] merge extension v9 into main (#8512)

AlexandreYang committed:
[corechecks/snmp] Handle snmp v1 no valid oids (#8613)

lebauce committed:
Allow setting DD_SOURCE header for new v2 logs intake (#8617)

vickenty committed:
Revert "[Serverless] merge extension v9 into main (#8512)" (#8625)

This reverts commit 7ca79bfc5537df9c1179cd190ac608979932716c.

Feature PR merged mid release freeze.

vickenty committed:
[release] add release entry for 7.30.0-rc.4 (#8626)

AlexandreYang committed:
[forwarder] Revert buffer size from 1000 to 100 (#8624)

* [forwarder] Revert buffer size

* Fix test

safchain committed:
Do not use zero cookie (#8627)

lebauce committed:
Default to v2 logs API for runtime security (#8628)

vickenty committed:
[release] add release entry for 7.30.0-rc.5 (#8634)

lebauce committed:
Specify the service as a log attribute (#8633)

brycekahle committed:
Make conntrack initialization timeout a configurable value (#8631)

p-lambert committed:
[networks] Check for `sockfd_lookup_light` symbol (#8640)

* [networks] Check for `sockfd_lookup_light` symbol

* [networks] Fix condition

vickenty committed:
[release] add release entry for 7.30.0-rc.6 (#8642)

Julio-Guerra committed:
pkg/trace: add new pkg/appsec package to proxy AppSec requests in trace-agent (#8359)

This change adds a new HTTP endpoint /appsec/proxy which reverse proxies requests to Datadog's AppSec Intake.

zippolyte committed:
Fix CloudFoundry tagger collector (#8644)

vickenty committed:
Revert "pkg/trace: add new pkg/appsec package to proxy AppSec requests in trace-agent (#8359)" (#8649)

This reverts commit c294b3a6dd9720baa478f2e805e2a8ca10fa154c.

zippolyte committed:
Fix CC API polling interval configuration, and allow batch size configuration (#8650)

zippolyte committed:
Use cloud_foundry config option to detect CF environment (#8648)

lebauce committed:
[runtime-security] enforce erpc read (#8654)

* Enforce eRPC bpf_probe_read result

* Fix path_id always zero

* Fix eRPC in-kernel execution check

* do not return path on mount error

Co-authored-by: Sylvain Afchain <safchain@gmail.com>

vickenty committed:
[release] add release entry for 7.30.0-rc.7 (#8656)

Commit:
[install_script] Fix APT keyring permissions (#8609)

albertvaka committed:
Suggest installing the IoT Agent on armv7l (#8591)

The previous code would fail later with a not-user-friendly error.

albertvaka committed:
Better detect if systemd is the init system (#8590)

Not only check if it's installed (ie: systemctl is there), but also that it's in use as the init process.

djmitche committed:
Rename master -> main in a few missed spots (#8622)

This fixes a few loose ends missed in 153d3e940cf77138d5a9ab821ed9351d534d14de.

ahmed-mez committed:
`agent-security` owns `compliance` (#8639)

julien-lebot committed:
Use custom python 2 (#8660)

brycekahle committed:
Reduce flakiness of some system-probe tests (#8608)

djmitche committed:
logs-agent: retry on 429's from intake (#8621)

When the intake server responds with an HTTP 429, the logs-agent now
retries sending the event using its usual backoff.  The v2 Event Intake
API will use 429's to signal that the server is overwhelmed.

brycekahle committed:
Fix nettop invoke task path (#8630)

Fixes #7747

vboulineau committed:
Fix cgroup parsing with systemd cgroup manager and CRIO (#8618)

vboulineau committed:
Fix Kubelet container ID matching in some scenarios (#8615)

vboulineau committed:
Fix delve build arguments (#8611)

clamoriniere committed:
Add new k8s manifest for kubernetes_state_core check (#8655)

* Update generate.sh k8s manifests script to support yq v4
* Add new k8s manifest for kubernetes_state_core check
* Fix Openshift SCC by adding NET_RAW capability

Julio-Guerra committed:
pkg/trace: add new pkg/appsec package to proxy AppSec requests in trace-agent (#8359) (#8670)

This change adds a new HTTP endpoint /appsec/proxy which reverse proxies requests to Datadog's AppSec Intake.

AlexandreYang committed:
[snmp] Suggest single quote for community string (#8659)

remeh committed:
Revert "[aggregator/ckey] implement a faster hash to reduce CPU and RAM usage (#8529)" (#8678)

This reverts commit 9eea12d089134fc43a3d5ce3d45c8ce4800a3b80.

julien-lebot committed:
Use custom python 3 on Windows (#8680)

paulcacheux committed:
[runtime-security] be more precise in the selinux cookbook dependency version spec (#8673)

paulcacheux committed:
[runtime-security] fix mutex lock around expired events (#8675)

* [runtime-security] fix mutex lock around expired events

* [runtime-security] be more clear about each mutex use for security server

brycekahle committed:
Ensure namespaced resolution has a non-zero namespace (#8683)

olivielpeau committed:
Set hostname to "none" by default for containerized CI Agent (#8652)

So that users don't have to set DD_HOSTNAME=none on top of
DD_INSIDE_CI=true

brycekahle committed:
Add kitchen testing of windows system-probe (#8585)

paulcacheux committed:
[CWS] small fixes (typos, test expected/actual order) (#8643)

* [runtime-security] fix typo in function name

* [runtime-security] fix tests to match expected/actual order

* [runtime-security] fix assert equal order in `pkg/security/probe/process_resolver_test.go`

* [runtime-security] fix assert equal order in `pkg/security/probe/reorderer_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/chmod_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/chown_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/chown32_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/link_test.go` `macros_test.go` and `mkdir_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/mount_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/overlayfs_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/probe_monitor_test.go`

* [runtime-security] fix assert equal order in `pkg/security/tests/process_test.go`

* [runtime-security] fix assert equal order in more files

* [runtime-security] fix assert equal order in `pkg/security/tests/open_test.go`

ahmed-mez committed:
[AD][Kubelet] AD checks should not target init containers (#8687)

Commit:
[install_script] Release version 1.6.0 (#8700)

safchain committed:
[runtime-security] fix timeout on functional tests on schema error (#8661)

* Make sure to not timeout on schema error

* Enforce exit time of exec event

Commit:
[install_script] Changelog for release 1.6.0 (#8703)

paulcacheux committed:
[runtime-security] fix iouring test (#8691)

* [runtime-security] upgrade iouring-go dependency

* [runtime-security] fix iouring close defer statement

* [runtime-security] add ubuntu 21.04 to kitchen tests, in order to test recent kernels

* [runtime-security] upgrade iouring-go dependency

djmitche committed:
[draft] Track agent binary sizes as well as package sizes (#8665)

* refactor pkg_metrics for readability

* add binary size metrics

djmitche committed:
Clone checkStats when fetched (#8668)

This avoids race conditions where the `checkStats` map might be updated
while it is still being read by the caller of `GetCheckStats` (which can
be `expvar`'s Handler).

vboulineau committed:
Allow Windows containers entrypoint executable to take command from environment variable (#8705)

ahmed-mez committed:
[KSMCore] Add owner tags to pod metrics (#8704)

maxday committed:
[Serverless] merge extension v9 into main (2) (#8662)

paulcacheux committed:
[runtime-security] fix the `TestProcessLineage` test (#8696)

* [runtime-security] fix the `TestProcessLineage` test

* [runtime-security] set the pid inside the process check

* [runtime-security] refactor a bit the `TestProcessLineage` test into `TestProcessExecExit`

* [runtime-security] simplify the exit pid check in the `TestProcessExecExit` test

safchain committed:
Ensure to use a mask instead of the raw event_type value (#8710)

brycekahle committed:
Move DNS instrumentation to a separate package (#8619)

AlexandreYang committed:
[corechecks/snmp] Support inline profile definition (#8679)

* [corechecks/snmp] Support inline profile definition

* Add reno

* remove unused test fixture

paulcacheux committed:
[runtime-security] check that the security-agent is initialized when handling a `/status` request (#8715)

* add nil check before getting AD errors

(cherry picked from commit 0385bcbcd41a2cc939f16d935f0a299f99c21f45)

* [runtime-security] add changelog

Co-authored-by: Celene Chang <celene@datadoghq.com>

paulcacheux committed:
[runtime-security] add missing serializers for SELinux events (#8720)

* [runtime-security] generate serializers for SELinux events

* [runtime-security] add "required" entry in SELinux JSON schema

* [runtime-security] fix SELinux schema

AlexandreYang committed:
[corechecks/snmp] Raise error on wrong sysobjectid (#8711)

ncracker committed:
Boyan env options in yaml (#8667)

* adding env var options

starting with the top three vars

* env description 

env and param descriptions should match

djmitche committed:
Add a checklist item for team label, milestone (#8726)

* Add checklist items for team label, milestone
* Suggest "Triage" milestone if milestone is unknown

Co-authored-by: Srdjan Grubor <sgnn7@sgnn7.org>

djmitche committed:
better sentence in PR template (#8727)

mbakht commented on pkg/process/checks/net.go:
```diff
@@ -260,16 +295,21 @@ func batchConnections(
+				EncodedDomains: dnsEncoder.EncodeDomainDatabase(namedb),
+			},
+			EncodedDnsLookups: dnsEncoder.EncodeMapped(batchDNS),
+			IndexedDomains:    batchDomainIndexes,
```
Do we need this for decoding?

mbakht replied:
Can you please explain why? Sorry - I may have missed it.

mbakht commented on pkg/process/checks/net_test.go:
```diff
@@ -239,15 +247,52 @@ func TestNetworkConnectionBatchingWithDomains(t *testing.T) {
+			assert.NotContains(t, domaindb, domains[1])
+
+			indexzero := indexOf(domains[0], domaindb)
+			indexdextwo := indexOf(domains[2], domaindb)
```
s/indexdextwo/indextwo?

mbakht commented on pkg/process/checks/net_test.go:
```diff
@@ -239,15 +247,52 @@ func TestNetworkConnectionBatchingWithDomains(t *testing.T) {
+			conn := connections.Connections[0]
+			val, ok := conn.DnsStatsByDomainByQueryType[0]
+			assert.True(t, ok)
+			assert.Equal(t, val.DnsStatsByQueryType[int32(network.DNSTypeA)].DnsTimeouts, uint32(1))
```
Can we have some other `DNSType` in the tests in addition to `DNSTypeA`?

hush-hush committed:
Allow 'telemetry.checks' to be set through env vars

hush-hush committed:
Remove the error returned by config.BindEnv

We never check the error and it's counter part BindEnvAndSetDefault
doesn't return an error either.

paulcacheux committed:
[runtime-security] simplify serializers (#8722)

paulcacheux committed:
[CWS] inject self-test rule/policy at startup and offer ways to trigger it (#8479)

* [runtime-security] add end-to-end injection rule

* [runtime-security] improve e2e injection

* [runtime-security] add config option for e2e injection test

* [runtime-security] rename functions/struct related to e2e => self-test

* [runtime-security] fix double import warning in pkg/sec/module/module.go

* [runtime-security] extract self tester to own file

* [runtime-security] extract self test open function

* [runtime-security] add more self test actual tests

* [runtime-security] small fix

* [runtime-security] add cobra command and grpc rpc for self-test

* [runtime-security] add self test error in the result message

* [runtime-security] fix go vet errors

* [runtime-security] rename configuration boolean for self test at start

* [runtime-security] add chown self test

* [runtime-security] add missing comment on `SelfTestFunctions`

* [runtime-security] append self test policies instead of doing a temp replace

* [runtime-security] simplify self test

* [runtime-security] fix self test cleanup and retry if needed

* [runtime-security] add missing comments and small fixes

* [runtime-security] fix golangci_lint error

* [runtime-security] use actual function to get rules from policy

* [runtime-security] use `AddPolicyVersion` for additionalPolicy

* [runtime-security] remove retry feature and fix underlying bug (event reuse ?)

* [runtime-security] omit self test policies from ruleset loaded events

* [runtime-security] add missing function documentation

* [runtime-security] add release note

* [runtime-security] offer option to completely disable the self tester (for tests mainly)

* [runtime-security] code improvement post review

* [runtime-security] change timeout to 3 seconds in self test

* [runtime-security] fix api generated file after rebase

* [runtime-security] add config for self test

* [runtime-security] embed self test config under self test config node

* [runtime-security] remove policiesDir argument of reload sub-function

* [runtime-security] move selftest specific code to self test module

* [runtime-security] move the self test actual implementation to self test module

* [runtime-security] move self test target file creation out of module start

* [runtime-security] remove the "run at startup" feature

paulcacheux committed:
[runtime-security]go generate protection check (#8732)

* [runtime-security] add test in Gitlab Pipeline to check that go generate on `pkg/security` has been run

* [runtime-security] apply review suggestions

ahmed-mez committed:
Revert Do not spawn tagger pull in a background goroutine (#8736)

sgnn7 committed:
[pkg/util] Add `hostname_file` configuration option (#8684)

In some environments, auto-detection of the hostname is not adequate and
environment variables cannot be used to set the value. In such cases, the
file on the host can also be used provide a file with the appropriate
value via `hostname_file` option.

Since getting the hostname via a file is pretty analogous to the rest of
the providers, it was made into a `HostnameProvider` and `options` were
added to the interface (and this provider may not be the only one that
could use parameterization). Ideally, we will expand this code in the
future to make the configuration file be a provider as well as there's no
need to treat the config as a bespoke provider.

Incidental to this PR some tests were added and minor formatting changes
made.

nhinsch committed:
[serverless] Try encrypting KMS key without encryption context first (#8729)

Commit:
[tasks] Add tasks for manipulating pipeline schedules (#8738)

Co-authored-by: Pablo Baeyens <pablo.baeyens@datadoghq.com>

nhinsch committed:
Make Serverless team owner of cmd/serverless directory (#8731)

safchain committed:
Use the first event type as index for the discarder mask (#8746)

djmitche committed:
Add comments about thread-safety of safeConfig (#8713)

djmitche committed:
Copy the list of source pointers in LogSources#GetSources (#8725)

This avoids the situation where the slice is modified while callers are
iterating over it, which may cause "missed" elements when removing
sources.

Co-authored-by: Brian Floersch <brian.floersch@datadoghq.com>

brycekahle committed:
Honor more config options in Windows network tracer (#8724)

mx-psi committed:
Add security_go_generate_check to JOBOWNERS (#8743)

AlexandreYang committed:
Upgrade gosnmp to 1.32 (#8426)

mx-psi committed:
Fix dependabot (replace with non-force-pushed commit) (#8744)

ahmed-mez committed:
[Docs] Update k8s manifests (#8749)

mx-psi committed:
Nudge to use `generate-licenses` if lint fails (#8753)

* Nudge to use `generate-licenses` if lint fails

- Add hint to use the `generate-licenses` task when `lint-licenses` fails.
- Update `generate-licenses

* Fix formatting

truthbk committed:
[dogstatsd] zstd support capture/replays (#8699)

* [dogstatsd] capture: support zstd captures

* [dogstatsd] capture: look at file version _after_ decompress

* [dogstatsd] capture: fix mmap release

* [dogstatsd] capture: make compression optional - defaults to true

* [reno] adding release note

* [dogstatsd] review feedback

* [licenses] adding zstd.

* [invoke] dogstatsd: static size has increased

dependabot[bot] committed:
Bump gotest.tools/gotestsum from 1.6.4 to 1.7.0 in /internal/tools (#8768)

Bumps [gotest.tools/gotestsum](https://github.com/gotestyourself/gotestsum) from 1.6.4 to 1.7.0.
- [Release notes](https://github.com/gotestyourself/gotestsum/releases)
- [Changelog](https://github.com/gotestyourself/gotestsum/blob/main/.goreleaser.yml)
- [Commits](https://github.com/gotestyourself/gotestsum/compare/v1.6.4...v1.7.0)

---
updated-dependencies:
- dependency-name: gotest.tools/gotestsum
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/mgechev/revive from 1.0.8 to 1.0.9 in /internal/tools (#8767)

Bumps [github.com/mgechev/revive](https://github.com/mgechev/revive) from 1.0.8 to 1.0.9.
- [Release notes](https://github.com/mgechev/revive/releases)
- [Changelog](https://github.com/mgechev/revive/blob/master/.goreleaser.yml)
- [Commits](https://github.com/mgechev/revive/compare/v1.0.8...v1.0.9)

---
updated-dependencies:
- dependency-name: github.com/mgechev/revive
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/vmihailenco/msgpack/v4 from 4.3.11 to 4.3.12 (#8379)

Bumps [github.com/vmihailenco/msgpack/v4](https://github.com/vmihailenco/msgpack) from 4.3.11 to 4.3.12.
- [Release notes](https://github.com/vmihailenco/msgpack/releases)
- [Commits](https://github.com/vmihailenco/msgpack/compare/v4.3.11...v4.3.12)

---
updated-dependencies:
- dependency-name: github.com/vmihailenco/msgpack/v4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/go-enry/go-license-detector/v4 in /internal/tools (#8766)

Bumps [github.com/go-enry/go-license-detector/v4](https://github.com/go-enry/go-license-detector) from 4.2.0 to 4.3.0.
- [Release notes](https://github.com/go-enry/go-license-detector/releases)
- [Commits](https://github.com/go-enry/go-license-detector/compare/v4.2.0...v4.3.0)

---
updated-dependencies:
- dependency-name: github.com/go-enry/go-license-detector/v4
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

davidor committed:
cluster_agent: use per-metric maxAge (#8735)

gbbr committed:
pkg/trace/sampler: improve ServiceCatalog tests (#8638)

piochelepiotr committed:
Upgrade sketches-go to v1.1.0 (#8733)

dependabot[bot] committed:
Bump github.com/shirou/gopsutil (#8769)

Bumps [github.com/shirou/gopsutil](https://github.com/shirou/gopsutil) from 3.21.5+incompatible to 3.21.7+incompatible.
- [Release notes](https://github.com/shirou/gopsutil/releases)
- [Commits](https://github.com/shirou/gopsutil/compare/v3.21.5...v3.21.7)

---
updated-dependencies:
- dependency-name: github.com/shirou/gopsutil
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ahmed-mez committed:
[KSMCore] Add pod.count and improve owner tagging (#8772)

dependabot[bot] committed:
Bump github.com/dgraph-io/ristretto from 0.0.3 to 0.1.0 (#8500)

* Bump github.com/dgraph-io/ristretto from 0.0.3 to 0.1.0

Bumps [github.com/dgraph-io/ristretto](https://github.com/dgraph-io/ristretto) from 0.0.3 to 0.1.0.
- [Release notes](https://github.com/dgraph-io/ristretto/releases)
- [Changelog](https://github.com/dgraph-io/ristretto/blob/master/CHANGELOG.md)
- [Commits](https://github.com/dgraph-io/ristretto/compare/v0.0.3...v0.1.0)

---
updated-dependencies:
- dependency-name: github.com/dgraph-io/ristretto
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* `inv generate-licenses`

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Pablo Baeyens <pablo.baeyens@datadoghq.com>

mx-psi committed:
Check also for go.sum diff (#8742)

paulcacheux committed:
[CWS] run kitchen functional tests on arm64, and fix bugs (#8651)

* [runtime-security] clarify that current security agent functional tests are running on x64

* [runtime-security] enable an ubuntu kitchen functional tests on arm64 for debug purposes

* [runtime-security] use chef version >= 15 for arm64

* [runtime-security] build functional tests in test_ebpf_arm64

* [runtime-security] run `chown32_test` only on amd64

* [runtime-security] add `ifSyscallSupported` to lchown tests

* [runtime-security] replace usage of syscall.SYS_OPEN, because the OPEN syscall is not available on arm64

* [runtime-security] do not compile stress testsuite from main branch

* [runtime-security] install i386 libc only on Intel x64

* [runtime-security] run arm64 functional testsuite under race detector

* [runtime-security] use more recent ubuntu 20.04 version (HWE) and test on 21.04

* [runtime-security] probe `security_bprm_check` when collecting args and envs

* [runtime-security] remove main compile version comparison

juliogreff committed:
[Tagger] Expire tags for stopped ECS EC2 tasks (#8758)

This issue was introduced in #6624 and identified and fixed for a
separate collector in #6780. The same story happened here: the ECS
collector did not have its own expiration logic (even though from a
cursory inspection it would look like so from the usage of
`taggerutils.Expire`), resulting in runaway memory usage over time for
nodes executing a lot of scheduled tasks, for instance.

Besides fixing this issue, this commit also changes all appropriate
collectors to use a slightly refactored version of `taggerutils.Expire`,
centralizing a little bit of behavior.

safchain committed:
Expose ctime and mtime to SECL (#8671)

juliogreff committed:
[Tagger] Better telemetry for queries and fetches (#8777)

* Add a `status` tag to the `tagger.queries` metric showing whether the
query succeeded, found no tags, or was called with an empty ID.

* Add a `tagger.fetches` metric that counts the number of fetches caused
by queries. This gives us better visibility into the amount of cache
misses in the tagger, when they error out, and when they find no tags.

juliogreff committed:
Correctly clear DockerUtil caches after listing containers (#8778)

The cache invalidation was wrong in two levels: it was clearing the
cache on every call to `dockerContainers` for the first 5 minutes (the
value of `invalidationInterval`) and then never again. It also never
updated `lastInvalidate`, so this has been very broken for a long time.

paulcacheux committed:
[runtime-security] add context for aptly replace in `go.mod` (#8747)

juliogreff committed:
[Tagger] Update telemetry for remote and replay tagger (#8783)

dependabot[bot] committed:
Bump github.com/DataDog/datadog-go (#8774)

Bumps [github.com/DataDog/datadog-go](https://github.com/DataDog/datadog-go) from 4.8.0+incompatible to 4.8.1+incompatible.
- [Release notes](https://github.com/DataDog/datadog-go/releases)
- [Changelog](https://github.com/DataDog/datadog-go/blob/master/CHANGELOG.md)
- [Commits](https://github.com/DataDog/datadog-go/compare/v4.8.0...v4.8.1)

---
updated-dependencies:
- dependency-name: github.com/DataDog/datadog-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

safchain committed:
Fix log level for invalidateDentry (#8785)

ahmed-mez committed:
[fix] Lower the log level for tagger pulling errors (#8787)

dependabot[bot] committed:
Bump golang.org/x/tools from 0.1.4 to 0.1.5 (#8792)

Bumps [golang.org/x/tools](https://github.com/golang/tools) from 0.1.4 to 0.1.5.
- [Release notes](https://github.com/golang/tools/releases)
- [Commits](https://github.com/golang/tools/compare/v0.1.4...v0.1.5)

---
updated-dependencies:
- dependency-name: golang.org/x/tools
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/spf13/cobra from 1.1.3 to 1.2.1 (#8570)

Bumps [github.com/spf13/cobra](https://github.com/spf13/cobra) from 1.1.3 to 1.2.1.
- [Release notes](https://github.com/spf13/cobra/releases)
- [Changelog](https://github.com/spf13/cobra/blob/master/CHANGELOG.md)
- [Commits](https://github.com/spf13/cobra/compare/v1.1.3...v1.2.1)

---
updated-dependencies:
- dependency-name: github.com/spf13/cobra
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

celenechang committed:
[ksm core] fix statefulset label mapping (#8789)

* fix statefulset label mapping

dependabot[bot] committed:
Bump github.com/aws/aws-sdk-go from 1.35.24 to 1.40.13 (#8780)

* Bump github.com/aws/aws-sdk-go from 1.35.24 to 1.40.13

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.35.24 to 1.40.13.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.35.24...v1.40.13)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Dustin J. Mitchell <dustin.mitchell@datadoghq.com>

derekwbrown committed:
review feedback

djmitche committed:
Include error in non-forced local flares (#8786)

When `agent flare` cannot contact the running agent, it falls back to a
local flare.  In this situation, it should include the error contacting
the agent in that local flare.

derekwbrown committed:
tests pass locally

derekwbrown committed:
Merge branch 'db/dedup-domains' of github.com:DataDog/datadog-agent into db/dedup-domains

derekwbrown commented:
Superceded by #8808 (rebase made a mess of this pr)
